src/test/java/org/folio/processing/mapping/functions/NormalizationFunctionTest.java

```diff
@@ -270,6 +271,81 @@ public void CAPITALIZE_shouldReturnExpectedResult() {
     }
   }
 
+  @Test
+  public void SET_DATE_TYPE_ID_shouldReturnExpectedResult() {
+    // given
+    String expectedInstanceDateTypeId = UUID.randomUUID().toString();
+    InstanceDateType firstInstanceDateType = new InstanceDateType()
+      .withId(UUID.randomUUID().toString())
+      .withName("No attempt to code")
+      .withCode("|");
+
+    InstanceDateType secondInstanceDateType = new InstanceDateType()
+      .withId(expectedInstanceDateTypeId)
+      .withName("Single known date/probable date")
+      .withCode("s");
+
+    InstanceDateType thirdInstanceDateType = new InstanceDateType()
+      .withId(UUID.randomUUID().toString())
+      .withName("Range of years of bulk of collection")
+      .withCode("k");
+
+    RuleExecutionContext context = new RuleExecutionContext();
+    context.setSubFieldValue("901214s19910101nyua");
+    context.setMappingParameters(new MappingParameters().withInstanceDateTypes(Arrays.asList(firstInstanceDateType, secondInstanceDateType, thirdInstanceDateType)));
+    // when
+    String actualInstanceDateTypeId = runFunction("set_date_type_id", context);
+    // then
+    assertEquals(expectedInstanceDateTypeId, actualInstanceDateTypeId);
+  }
+
+  @Test
+  public void SET_DATE_TYPE_ID_shouldReturnUnspecifiedIssuanceModeIdIfNoMatchedExists() {
```

Comment:
```suggestion
  public void SET_DATE_TYPE_ID_shouldReturnNoAttemptToCodeIfNoMatchedExists() {
```

Reply:
@RuslanLavrov  Thank you! That was a TYPO.